LiberateAutomation/src/com/liberate/automation/pom/MaintainDepositReason.java

```diff
@@ -47,4 +52,109 @@
 	// New record creation_Maintain Deposit Reason - Command button
 	By Acceptbutton = By.xpath("//div[@class='icePnlGrp iceNormalPnlGrd']//input[@value='Accept']");
 	By CancelButton = By.xpath("//div[@class='icePnlGrp iceNormalPnlGrd']//input[@value='Cancel']");
+	
+	/***
+	 * Default constructor that takes TestAction object as argument.
+	 * 
+	 * @param action
+	 *            The TestAction class object, which will be used to run test
+	 *            steps
+	 */
+	public MaintainDepositReason(TestActions action) {
+		this.action = action;
+	}
+	
+	/***
+	 * The method to navigate to Maintain Deposit Reason screen
+	 * 
+	 * @return Returns true if able to navigate, else will return false.
+	 */
+	public boolean navigate() {
+		boolean passed = false;
+
+		passed = action.waitFor(LiberateCommon.LevelOne.Orders, 4, true);
+		passed = action.clickOn(LiberateCommon.LevelOne.CreditControl);
+		passed = action.waitFor(LiberateCommon.CreditControl.DepositMaintenance, 4, true);
+		passed = action.clickOn(LiberateCommon.CreditControl.DepositMaintenance);
+		passed = action.waitFor(LeftLink.DepositMaintenance.MaintainDepositReason, 4, true);
+		passed = action.clickOn(LeftLink.DepositMaintenance.MaintainDepositReason);
+
+		return passed;
+	}
+
+	
+	
+	/***
+	 * Method to Click on'New' command button to navigate to Maintain Deposit Reason - New Functionality
+	 * 
+	 * @return True, if able to select; false if not able to select.
+	 */
+	public boolean clickNewButton() {
+		boolean passed = false;
+
+		passed = action.waitFor(NewButton, 4, true);
+		passed = action.clickOn(NewButton);
+
+		return passed;
+	}
+	
+	
+	/***
+	 * Method to create New Maintain Deposit Reason Functionality
+	 * 
+	 * @return True, if able to select; false if not able to select.
+	 */
+	public boolean maintainDepositReason_CreateNew(String DepositReasonCode, String Description, String InterestToBeCreditedToCustomersBill,

```

Comment:
Method names are not accurate. 
1. Aleady class name is MaintainDepositReason, hence you can remove that from method name.
2. Method Name should be an Action and it should be clear. Like 'createNewDepositeReason' or 'clickOnNewButton' or' clickOnResetButton'